### PR TITLE
docs/cephadm: fix broken links in cephadm docs

### PR DESCRIPTION
**Pull request by shrutiparvekar**
In the documentation https://docs.ceph.com/en/quincy/cephadm/services/osd/, there is a broken link in "List Devices" chapter. This change fixes the documentation to point to the correct link https://docs.ceph.com/en/quincy/rados/operations/devices
Tested that docs build is successful after the change as per https://github.com/ceph/ceph#building-the-documentation

Fixes: https://tracker.ceph.com/issues/55763

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by anthonyeleven** [comment]
I think we probably shouldn't link to a specific release's page.  @zdover23 ?

**Comment by ljflores** [comment]
@anthonyeleven FYI this is a contribution from Grace Hopper Open Source Day!

**Comment by zdover23** [comment]
> I think we probably shouldn't link to a specific release's page. @zdover23 ?

@shrutiparvekar, @anthonyeleven: Anthony is correct that this kind of change should be handled a bit differently. I take it that this is part of the Grace Hopper project, so I am going to merge it and backport it (to show how the workflow works) and then I'll create a new issue showing how to do this so that the change is durable and so that it refers to the RST files and not to the rendered documentation.

**Comment by zdover23** [comment]
@shrutiparvekar: I had a look at this, and it looks like you can remove this line entirely and practice squashing (interactive rebasing):

>[zdover@fedora ceph]$ grep -ir "_device management" *
>build-doc/output/html/_sources/cephadm/services/osd.rst.txt:.. _device management: ../rados/operations/devices
>grep: build-doc/doctrees/cephadm/services/osd.doctree: binary file matches
>doc/cephadm/services/osd.rst:.. _device management: ../rados/operations/devices

> [zdover@fedora services]$ git log -S "_device management" osd.rst
> commit 8c70398949f773d8b992ad8ae6c71383460d2932
> Author: Sebastian Wagner <swagner@redhat.com>
> Date:   Mon Sep 13 17:15:33 2021 +0200
> 
>     doc/cephadm: move services into services/
>     
>     This is going to clean up the toctree a bit.
>     
>     Signed-off-by: Sebastian Wagner <sewagner@redhat.com>


This link just isn't used anywhere in the docs, apparently. (Update, from three minutes in the future of that sentence's composition: Yep. Sebastian Wagner carried these links over from an old structure in which services.rst became services/[various files].rst.). So we can confidently prune this link from the docs.

Here are my instructions for squashing: https://docs.ceph.com/en/quincy/start/documenting-ceph/#squash-extraneous-commits

If you have command of the necessary leisure to write up your complaints about this squashing procedure, I'd be grateful and would credit you as a co-author of any changes I make to the docs in response to your complaints.

**Comment by zdover23** [comment]
@shrutiparvekar: I know that I just wrote an awful lot of text in your direction. Here's the upshot: keep finding broken links like this, and do exactly what you did here, and I'll make sure that they get fixed. Thanks for finding this. Good looking out.

File: doc/cephadm/services/osd.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ***********
 OSD Service
 ***********
-.. _device management: ../rados/operations/devices
+.. _device management: https://docs.ceph.com/en/quincy/rados/operations/devices
```
Comment (ljflores) [inline doc/cephadm/services/osd.rst:4]: Hey @shrutiparvekar congrats on your first PR! It seems like some of our doc writers (Zac and Anthony) have some feedback. I'm taking a look on my end..

Comment (zdover23) [reply doc/cephadm/services/osd.rst:4]: @ljflores: Sorry for dumping that wall of text on you guys. I've thought about it, and finding these dead links is useful. If this is a useful exercise for Grace Hopper day, keep doing exactly this and I will take care of all the tedious business of situating @shrutiparvekar's discoveries in the Procrustean bed of our CI workflow.

**Comment by shrutiparvekar** [comment]
Hi @zdover23 Thanks for your inputs!
As I understand, I see two options:
1. Change the link to point to ../../../rados/operations/devices (This is the correct relative path)
2. Delete the device management link altogether. In this case, we would also need to edit the text where this link is referenced to remove it.

Would you recommend going with option 2?
(Thanks for sharing the instructions for squashing)

**Comment by zdover23** [comment]
> Hi @zdover23 Thanks for your inputs! As I understand, I see two options:
> 
>     1. Change the link to point to ../../../rados/operations/devices (This is the correct relative path) to avoid having the release-specific link
> 
>     2. Delete the device management link altogether. In this case, we would also need to edit the text where this link is referenced to remove it.
> 
> 
> Would you recommend going with option 2? (Thanks for sharing the instructions for squashing)

I actually found one place in doc/cephadm/services/osd.rst where _device management is referred to: line 83, which reads as follows:

     information about interacting with these LEDs, refer to `device management`_. 

My plan, which you are one-hundred percent welcome to undertake with as much or as little help as you want from me, is to put an internal link in the document as described here: https://docs.ceph.com/en/reef/start/documenting-ceph/#internal-links

I'm sorry if all of this is overwhelming. It took me two years to learn all this stuff, and I can only imagine that it might be daunting.

**Comment by zdover23** [comment]
@ljflores: I'll do whatever you decide is best.

**Comment by ljflores** [comment]
> @ljflores: I'll do whatever you decide is best.

@shrutiparvekar can look into adding an internal link. I think this is great contribution experience!

@zdover23 you can help out just in case Shruti has further questions!

**Comment by shrutiparvekar** [comment]
Okay, that sounds good. I will start looking into adding an internal link and will update the PR soon.

**Comment by ljflores** [comment]
@shrutiparvekar by the way, @zdover23 is the documentation lead for Ceph! @anthonyeleven is also a major documentation contributor. If you keep contributing to Ceph, you will see their names pop up a lot!

**Comment by anthonyeleven** [comment]
Thanks, Laura.  And we love to help new contributors ramp up.

**Comment by ljflores** [comment]
@shrutiparvekar let us know if you need any further help!

**Comment by ljflores** [comment]
@shrutiparvekar do you have any questions on this PR? If so, let us know! We'd be happy to help.

**Comment by shrutiparvekar** [comment]
Hi, I have updated the PR to use an internal link to fix this issue.

> By the way, a pretty cool thing about documentation is that the PR generates a version based on your PR: https://ceph--53601.org.readthedocs.build/en/53601/

Thanks for sharing this, @ljflores. This is really helpful!

**Review by ljflores — COMMENTED** [review]
Hey @shrutiparvekar, thanks for updating!

A couple of things:

Your commit should have a title like your PR does. You can visit your local branch and type:
```
git commit --amend
```
to edit the commit, and add `docs/cephadm: fix broken links in cephadm docs` as the commit title.

Here's an example of a commit I made recently so you can see what the structure should look like:
```
commit 9c5755a4ac55ef6b825603eea52785f936888906
Author: Laura Flores <lflores@ibm.com>
Date:   Mon Oct 9 20:24:14 2023 +0000

    qa/suites/upgrade: fix env indentation in stress-split upgrade tests
    
    This is an issue with the stress-split yaml files, as introduced in https://github.com/ceph/ceph/pull/51889.
    
    The stress-split tests have an incorrectly-intented "env" section, which teuthology detects as an entry for "clients".
    
    Fixes: https://tracker.ceph.com/issues/63158
    Signed-off-by: Laura Flores <lflores@ibm.com>
```

Also, it looks like the docs failed to build, due to a syntax error. If you see the automated CI checks, you can see a red X next to the tests that failed:
image

You can click on "Details" --> "Console log" to see why it failed:
```
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/cephadm/services/osd.rst:80: WARNING: Mismatch: both interpreted text role prefix and reference suffix.
looking for now-outdated files... none found
```

@zdover23 @anthonyeleven perhaps you have some ideas for how to fix it?

**Review by ljflores — COMMENTED** [review]
Also, you can check the rendering of the documentation on this link: https://ceph--53601.org.readthedocs.build/en/53601/cephadm/services/osd/

The link you changed still doesn't lead to the right page, so you can use that to verify your work!

File: doc/cephadm/services/osd.rst [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ***********
 OSD Service
 ***********
-.. _device management: ../rados/operations/devices
```
Comment (ljflores) [inline doc/cephadm/services/osd.rst:4]: I'm pretty sure the fix would be to write

`../../rados/operations/devices`

as opposed to

`../rados/operations/devices`

Comment (ljflores) [reply doc/cephadm/services/osd.rst:4]: Ah, I see @zdover23 advised to add an internal link as described in https://docs.ceph.com/en/reef/start/documenting-ceph/#internal-links, which is what the current fix is attempting. Perhaps you have further advice on this method @zdover23 ?

**Pull request closed without merging**